facade/src/main/java/org/jboss/pnc/facade/impl/BrewPusherImpl.java

```diff
@@ -251,7 +251,7 @@ private BuildRecord getLatestSuccessfullyExecutedBuildRecord(Base32LongID buildR
             // if status is NO_REBUILD_REQUIRED, find the last BuildRecord with status SUCCESS for the same idRev.
             IdRev idRev = buildRecord.getBuildConfigurationAuditedIdRev();
             BuildRecord latestSuccessfulBuildRecord = buildRecordRepository
-                    .getLatestSuccessfulBuildRecord(idRev, buildRecord.isTemporaryBuild());
+                    .getAnyLatestSuccessfulBuildRecordWithRevision(idRev, buildRecord.isTemporaryBuild());
```

Comment:
I think it's better to use ` buildRecord.getNoRebuildCause();`
```java
            BuildRecord noRebuildCause = buildRecord.getNoRebuildCause();
            if (noRebuildCause != null) {
                return noRebuildCause;
            } else {
                String message = "There is no SUCCESS build before NO_REBUILD_REQUIRED.";
                log.error(message);
                throw new InconsistentDataException(message);
            }
```

Reply:
ah nice one

Reply:
Please remove the comment `                // In case of temporary builds it can happen. The SUCCESS build might be already garbage-collected.`, it's not true anymore